src/xo-app/pool/tab-patches.js

```diff
@@ -1,13 +1,160 @@
-import _ from 'intl'
+import ActionButton from 'action-button'
+import Component from 'base-component'
 import React from 'react'
+import SortedTable from 'sorted-table'
+import TabButton from 'tab-button'
+import _ from 'intl'
+import forEach from 'lodash/forEach'
+import isEmpty from 'lodash/isEmpty'
+import map from 'lodash/map'
 import { Container, Row, Col } from 'grid'
+import { connectStore } from 'utils'
+import {
+  createFilter,
+  createGetObjectsOfType
+} from 'selectors'
+import {
+  getHostMissingPatches,
+  installAllHostPatches
+} from 'xo'
+
+// ===================================================================
+
+const MISSING_PATCHES_COLUMNS = [
+  {
+    name: _('srHost'),
+    itemRenderer: host => host.name_label,
+    sortCriteria: host => host.name_label
+  },
+  {
+    name: _('hostDescription'),
+    itemRenderer: host => host.name_description,
+    sortCriteria: host => host.name_description
+  },
+  {
+    name: _('hostMissingPatches'),
+    itemRenderer: (host, missingPatches) => missingPatches[host.id],
+    sortCriteria: (host, missingPatches) => missingPatches[host.id]
+  },
+  {
+    name: _('patchUpdateButton'),
+    itemRenderer: host => (
+      <ActionButton
+        btnStyle='primary'
+        handler={installAllHostPatches}
+        handlerParam={host}
+        icon='host-patch-update'
+      />
+    )
+  }
+]
+
+@connectStore(() => {
+  const getHosts = createGetObjectsOfType('host').filter(
+    (_, props) => host => props.pool.id === host.$pool
+  )
+
+  return {
+    hosts: getHosts
+  }
+})
```

Comment:
Yes it works but because the selector is shared between all instances of the component it can creates issues with its cache.

The rule of thumb is to never share a selector obtained via a `create*` method.